### PR TITLE
tests: add standalone tests module for testing generated Go code

### PR DESCRIPTION
**Pull request by ydnar**
Implement a lightweight pattern for testing generated Go code using TinyGo + `wasip2`. In addition, the `testdata` directory now has an empty `go.mod` file to exclude it from module distribution.

- Generated Go interfaces can be rebuild with `make generate` from the root directory.
- To run tests, `cd tests && tinygo test -v -target=./wasip2.json ./...`

### CODE REVIEW

**Comment by ydnar** [comment]
This requires TinyGo >= 0.33.1 to pass, which is not yet available.

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -170,5 +170,9 @@ jobs:
         if: ${{ matrix.tinygo-version != '0.32.0' }}
         run: tinygo test -v -target=wasip2 ./...
 
+      - name: Test generated Go with TinyGo >= 0.33.0
+        if: ${{ matrix.tinygo-version != '0.32.0' }}
+        run: cd tests && tinygo test -v -target=./wasip2.json ./...
```
Comment (Mossaka) [inline .github/workflows/test.yaml:175]: add this command to Makefile so we can do `make test`

File: tests/generated/wasi/clocks/monotonic-clock/empty.s [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// This file exists for testing this package without WebAssembly,
```
Comment (Mossaka) [inline tests/generated/wasi/clocks/monotonic-clock/empty.s:1]: We prohably don't want to commit the generated Go code?

File: tests/wasip2.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+	"inherits": ["wasip2"],
+	"emulator": "wasmtime --wasm component-model -Sinherit-network -Sallow-ip-name-lookup --dir={tmpDir}::/tmp {}"
```
Comment (Mossaka) [inline tests/wasip2.json:3]: For runtime tests that don't require networking, can we customize this JSON to suit specific test cases?

File: tests/generated/wasi/clocks/monotonic-clock/empty.s (hunk 2 above)
Comment (ydnar) [reply tests/generated/wasi/clocks/monotonic-clock/empty.s:1]: I think it's useful to have the generated code committed for gopls/IDE and to show how generated code changes over time.

File: tests/wasip2.json (hunk 3 above)
Comment (ydnar) [reply tests/wasip2.json:3]: I submitted a PR to TinyGo to make this unnecessary.

Comment (ydnar) [reply tests/wasip2.json:3]: https://github.com/tinygo-org/tinygo/pull/4468

**Comment by ydnar** [comment]
This now works with TinyGo 0.34.0!

**Pull request merged**